packages/flame/lib/src/collisions/hitboxes/circle_hitbox.dart

```diff
@@ -50,10 +50,19 @@ class CircleHitbox extends CircleComponent with ShapeHitbox {
     _temporaryCenter
       ..setFrom(_temporaryAbsoluteCenter)
       ..sub(ray.origin);
-    _temporaryCenter.projection(ray.direction, out: _temporaryLineSegment.to);
-    _temporaryLineSegment.to
-      ..x *= (ray.direction.x.sign * _temporaryLineSegment.to.x.sign)
-      ..y *= (ray.direction.y.sign * _temporaryLineSegment.to.y.sign);
+
+    if (_temporaryCenter.isZero()) {
+      // If _temporaryCenter is zero, it's projection onto ray.direction
+      // will be zero. In that case, directly use ray.direction as temp
+      // end point of line segment.
+      _temporaryLineSegment.to.setFrom(ray.direction);
```

Comment:
I don't understand how this only affected the rays going straight north, feels like it should have affected all rays cast directly from a circle center?

Reply:
Yeah, it was affecting all ray. I must have missed something when I was reporting the issue back then. But today when I tried casting rays in any direction, it was showing the same bug. Coincidence of ray origin and hitbox's center is the only condition for this to happen.